lib/Response.php

```diff
@@ -64,10 +64,53 @@ public function body()
     /**
      * The response headers
      *
+     * @param bool $assoc
+     *
      * @return array
      */
-    public function headers()
+    public function headers($assoc = false)
     {
-        return $this->headers;
+        if (!$assoc) {
+            return $this->headers;
+        }
+        
+        return $this->prettifyHeaders($this->headers);
+    }
+    
+    /**
+      * Returns response headers as associative array
+      * 
+      * @param array $headers
+      *
+      * @return array
+      * 
+      * @throws \InvalidArgumentException
+      */
+    private function prettifyHeaders($headers)
+    {
+        if (!is_array($headers)) {
+            throw new \InvalidArgumentException('$headers should be array');
+        }
+
+        return array_reduce(
+            array_filter($headers),
+            function ($result, $header) {
+                if (empty($header)) {
+                    return $result;
+                }
+
+                if (false === strpos($header, ':')) {
```

Comment:
Why header not name? Can you provider description this case?
I think is dirty  hack